build.ps1

```diff
@@ -0,0 +1,290 @@
+
+    $env:PATH = $env:PYTHONPATH = $($($paths | foreach { ";${condaPath}\$_".Trim() }) -join ('')) + ";C:\Windows\System32\" <# ¯\_(ツ)_/¯ #>
+    if ($debug) { echo $env:PATH }
+
+    if (!(test-path -path $condaPath)) {
+        try {
+            $progressPreference = 'silentlyContinue'
+
+            echo "Downloading conda"
+            $condaInstallerURL = "https://repo.anaconda.com/miniconda/Miniconda3-latest-Windows-x86_64.exe"
+            $condaInstaller = "$env:TMP\\Miniconda3-latest-Windows-x86_64.exe"
+
+            invoke-webrequest $condaInstallerURL -outfile $condaInstaller
+
+            echo "Installing conda"
+            $progressPreference = 'Continue'
+
+            $condaArgs = "/InstallationType=JustMe /RegisterPython=0 /AddToPath=0 /S /D=${condaPath}"
+            Start-Process $condaInstaller $condaArgs -Wait
+        }
+        catch {
+            $_
+            echo "Error installing conda"
+            exit 1a
+        }
+    }
+
+    conda update conda -y
+
+    $condaVer = conda --version
+    if ($debug) { echo "$condaVer" }
+    if (!($condaVer -match "4.1*.*")) {
+        echo "Could not locate conda executable"
+        exit 1
+    }
+
+    conda config --add channels conda-forge
+    conda config --set channel_priority strict
+
+    $result = conda info --env | findstr $tmpEnv
+    if ($debug) { echo "Found env: ${result}" }
+
+    if ( [string]::IsNullOrEmpty($result)) {
+        echo "Creating conda environment"
+        $envStr = "conda create -n ${tmpEnv} -y python=3.9 libpython-static"
+        iex $envStr
+    }
+    else {
+        echo "Conda environment already exists"
+    }
+
+    $activate = "conda activate ${tmpEnv}"
+    iex $activate
+
+    $instStr = "conda install -c conda-forge -y $($libraries | % { "$_" })"
+    iex $instStr
+
+    if ($debug) { echo "PATH: $env:PATH" }
+    if ($debug) { echo "PYTHONPATH: $env:PYTHONPATH" }
+
+    $version = $(python --version).replace('Python ', '')
+}
+
+# --| Begin Build Process ----------------------------
+$installer = ''
+
+# --| PyInstaller ------------------------------------
+function BuildPyInstaller() {
+    $req = Get-Content requirements.txt
+    $imports = @('glcontext', 'openimageio', 'wx', 'numpy', 'opencv-python', 'gswidgetkit', 'gsnodegraph')
+    $tmp = $env:TEMP
+
+    $installer = 'pyinstaller'
+    $args = [System.Collections.ArrayList]@()
+    # [void]$args.Add('--upx-dir .\upx')
+    [void]$args.Add("${installer} ")
+    # [void]$args.Add("--onefile ")
+    [void]$args.Add("--noconfirm")
+    [void]$args.Add("-n GimelStudio")
+    [void]$args.Add("$(PyInstallAddData)")
+    [void]$args.Add("--workpath '${tmp}'")
+    [void]$args.Add("--runtime-tmpdir ${tmp}")
+    [void]$args.Add("--exclude-module=.git")
+    [void]$args.Add("-i assets/GIMELSTUDIO_ICO.ico")
+    [void]$args.Add("$($req | % { "--hidden-import $_".Trim() })")
+    [void]$args.Add("$($imports | % { "--hidden-import $_".Trim() })")
+    [void]$args.Add("src/main.py")
```

Comment:
Add ``main.py`` inside of the ``src`` directory (see step 4 on the [README](https://github.com/GimelStudio/GimelStudio#windows)). Using what you have here may create path errors (as Gimel Studio uses relative paths e.g. [here](https://github.com/GimelStudio/GimelStudio/blob/master/src/gimelstudio/config.py#L75))

Reply:
I don't quite understand what you are requesting. I am doing the same thing as the current script. See here: https://github.com/GimelStudio/GimelStudio/blob/0adccf685bd7942de83282f222d9e6bf25a00555/build.py#L379
Both PyInstaller and Nuitka require relative paths from the project root/build script location in the build arguments.